### PR TITLE
Refactor Chef::Mixin::WideString to remove implicit Windows dependency.

### PR DESCRIPTION
**Pull request by mcquin**
* Renames `'lib/chef/wstring.rb'` to `'lib/chef/wide_string.rb'` to match mixin name.
* Moves `wide_to_utf8` and `utf8_to_wide` methods from Windows unicode files into mixin.

### CODE REVIEW

File: lib/chef/win32/unicode.rb [hunk 1]
```diff
@@ -43,13 +46,7 @@ def read_wstring(num_wchars = nil)
         num_wchars = length
       end
 
-      Chef::ReservedNames::Win32::Unicode.wide_to_utf8(self.get_bytes(0, num_wchars*2))
+      wide_to_utf8(self.get_bytes(0, num_wchars*2))
     end
   end
 end
-
-class String
```
Comment (jaym) [inline lib/chef/win32/unicode.rb]: not sure if you can remove this

**Comment by mcquin** [comment]
Windows is going to fail so hard. HAHA. Will ping when ready for review.

**Comment by mcquin** [comment]
@chef/client-windows this is ready for review.

**Comment by chefsalim** [comment]
Nice!

**Comment by jaym** [comment]
:+1:

**Comment by btm** [comment]
:+1:

**Pull request merged**